src/components/snackbar/index.tsx

```diff
@@ -0,0 +1,51 @@
+import Stack from "@mui/material/Stack";
+import Snackbar from "@mui/material/Snackbar";
+import MuiAlert, { AlertColor, AlertProps } from "@mui/material/Alert";
+import { forwardRef, SyntheticEvent, useEffect, useState } from "react";
+import { CoreModule, getCoreModuleRoute } from "lib/router";
+import { useNavigate } from "react-router-dom";
+import { Backdrop } from "@mui/material";
+
+const Alert = forwardRef<HTMLDivElement, AlertProps>(function Alert(props, ref) {
+  return <MuiAlert elevation={6} ref={ref} variant="filled" {...props} />;
+});
+
+interface SnackbarProps {
+  message: string;
+  severerity: AlertColor;
+  isOpen: boolean;
+  navigateTo: CoreModule;
+}
+
+export const CustomizedSnackbar = ({ message, severerity, isOpen, navigateTo }: SnackbarProps) => {
+  const [open, setOpen] = useState(isOpen);
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    setOpen(isOpen);
+  }, [isOpen]);
+
+  const handleClose = (event?: SyntheticEvent | Event, reason?: string) => {
+    if (reason === "clickaway") {
+      return;
+    }
+
+    setOpen(false);
+    if (severerity === "success") {
+      navigate(getCoreModuleRoute(navigateTo));
+    }
+  };
+
+  return (
+    <Backdrop open={open}>
+      <Stack spacing={2} sx={{ width: "100%" }}>
+        <Snackbar open={open} autoHideDuration={2000} onClose={handleClose}>
+          <Alert onClose={handleClose} severity={severerity} sx={{ width: "100%" }}>
```

Comment:
Using sx prop together with styled components leads to mess. Let's keep only way to style components and keep all css related code in styled.ts(x) files?

Reply:
Yeah, This is noted. But here actually this sx is not needed anyway so removed it.